src/Component.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\twig_fractal\Component.
+ */
+
+namespace Drupal\twig_fractal;
+
+use Drupal\Component\Utility\Html;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Prepares context variables for a Twig `render` node.
+ *
+ * Default component variables are recieved from a Fractal component definition
+ * file and merged with the existing `render` node context variables.
+ */
+class Component {
+
+  /**
+   * The pathname of the component to render.
+   *
+   * @var string
+   */
+  protected $pathname;
+
+  /**
+   * The name of the component to render.
+   *
+   * @var string
+   */
+  protected $name;
+
+  /**
+   * The names of the variants to render.
+   *
+   * @var array
+   */
+  protected $variants = [];
+
+  /**
+   * Constructs component class properties from a given Twig file handle.
+   */
+  public function __construct($handle) {
```

Comment:
Missing PHPDoc for `$handle`.

1. Is it possible to type-hint the data type?

2. The description should clarify what kind of values can be passed.